src/sql/workbench/services/notebook/common/notebookService.ts

```diff
@@ -68,6 +68,25 @@ export interface INotebookService {
 	getMimeRegistry(): RenderMimeRegistry;
 
 	renameNotebookEditor(oldUri: URI, newUri: URI, currentEditor: INotebookEditor): void;
+
+	/**
+	 * Checks if a notebook has previously been marked as trusted, and that
+	 * the notebook has not changed on disk since that time. If the notebook
+	 * is currently dirty in the app, the previous trusted state will be used even
+	 * if it's altered on disk since the version in our UI is based on previously trusted
+	 * content.
+	 * @param notebookUri the URI identifying a notebook
```

Comment:
I guess my main question is this... What's the behavior if you save a notebook, trust it, then with another notebook, save it as the same URI? It would then be trusted, right?

This may be a corner case, just wanted to call it out 😄